File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length-1;
        int[] result = new int[n+1];
        int start =0, end = n;
        
        while(start<=end)
        {
            int squareStart = nums[start] * nums[start];
            int squareEnd = nums[end] * nums[end];
           if(squareStart>=squareEnd)
           {
               result[n--] =squareStart;
               
               start++;
           }
           else
           {
             result[n--] = squareEnd;
             end--;
            

           }
        }
        return result;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
   // TC = O(n) SC- O(n)
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(n,0);
        int left =0;
        int right = n-1;
        for(int i = n-1;i>=0;i--)
        {
            if(abs(nums[left])>abs(nums[right]))
            {
                result[i]= nums[left] * nums[left];
                left++;
              
            }
            else
            {
                result[i]= nums[right] * nums[right] ;
                right--;
            }

        }
        return result;
    }
};